src/test/bench/testfilestore_backend.cc

```diff
@@ -57,9 +57,10 @@ void TestFileStoreBackend::write(
 
   os->queue_transaction(
     osr,
-    t,
+    std::move(*t),
```

Comment:
> Making the rvalue reference is a promise that no other references exist

Not really, no. The call to `queue_transaction()` will move data out of the Transaction at memory address `t` (and it does this synchronously), so there's no race between the move and the deletion of `t`.

But you are correct that the `delete t;` added below is a bug, because `C_DeleteTransWrapper` will already delete it. `t` should just be on the stack here, so that both `delete t;` and the `C_DeleteTransWrapper` are unnecessary.